### ISSUE TITLE
Benchmark identity churn

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#8063]
Assess the impact of identity churn upon operation of Cilium.

* At least one local endpoint
* Apply policy that selects the endpoint, allowing traffic to/from identities with a given label
  * Initially inject 1 rule, can later attempt a higher baseline number
* Inject a baseline number of unique identities into the kvstore
* Vary the number of identities in the kvstore at a particular rate
  * Test 1: Set the baseline + varied identities to *not be* selected by the policy rule
  * Test 2: Set the baseline + varied identities to *be* selected by the policy rule
* Observe {CPU consumed, policy generation time}

**Comment by joestringer** [comment]
Target pod:
```
apiVersion: v1
kind: Pod
metadata:
  name: pause
  labels:
    name: pause
spec:
  containers:
  - name: pause
    image: "gcr.io/google_containers/pause"
    ports:
    - containerPort: 80
```

**Comment by joestringer** [comment]
Simple identity injection (appears to support 10-15 identity updates per second, decreasing with scale of number of identities):

1. Log into Cilium pod
1. Inject 8000 unique identities
```
$ for i in $(seq 0 7999); do cilium kvstore set --key "cilium/state/identities/v1/id/$((65536+$i))" --value "k8s:select=false;k8s:test-id=$i"; done
```

NB: `select:false` means that the policy above will *not* select these identities (Test 1). Set to "true" to have the policy select those identities (Test 2).

**Comment by ianvernon** [comment]
Will document steps here about how to get this working on the Cilium Vagrant setup, specifically port forwarding for metrics to be viewable in browser on my local host.

**Comment by joestringer** [comment]
The test (1) and test (2) do not appear to have a significant difference in policy calculation time.

I ran 8 endpoints, 7 of which had policy. At least one had a policy that would select the identities being added.

I scaled up to 20K identities without incident. CPU usage reported through the standard cilium prometheus/grafana setup stayed quite low - around 5-10%. There were around 15-16 kvstore operations per second over the test period.

As long as nothing else was going on on my laptop at the time, even 99th percentile endpoint regeneration appeared to be well under 50ms. In worst case, headerfile prep often shows up as a meaningful proportion of the regeneration time (See also #8144). 90th percentile was typically around 20ms.
